### PR TITLE
Move security.txt to wagtail.org

### PR DESCRIPTION
**Pull request by RealOrangeOne**
_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

RTD is a little too magical for us, and the current `security.txt` in its current location doesn't work correctly - it's served from `/en/stable/.well-known/security.txt` instead. I've moved the security.txt to be served by wagtail.org directly (and thus is applicable to all Wagtail projects).

This also removes some duplication with the `SECURITY.md`.

I've setup a redirect to make the URL work for now, which I'll update to point to wagtail.org once it's deployed.

Depends on https://github.com/wagtail/wagtail.org/pull/355/

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: docs/contributing/security.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Reporting security issues
 
+```{warning}
+Ensure you are viewing our [latest security policy](https://docs.wagtail.org/en/latest/contributing/security.html).
```
Comment (RealOrangeOne) [inline docs/contributing/security.md:4]: I couldn't find a nicer way to reference the canonical latest version of a page. If there's some sphinx shorthand I don't know about, please tell me!

File: docs/conf.py [hunk 2]
```diff
@@ -171,7 +171,7 @@
 # Add any extra paths that contain custom files (such as robots.txt or
 # .htaccess) here, relative to this directory. These files are copied
 # directly to the root of the documentation.
-html_extra_path = ["public"]
+html_extra_path = ["robots.txt"]
```
Comment (gasman) [inline docs/conf.py:174]: @RealOrangeOne Is this valid? The [sphinx docs](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path) say

> A list of **paths that contain extra files** not directly related to the documentation, such as robots.txt or .htaccess.

which I'd interpret as meaning the files _within_ these paths are copied to the root - obviously `robots.txt` doesn't have files within it. Any reason this shouldn't remain unchanged (with robots.txt left in public/)?

Comment (RealOrangeOne) [reply docs/conf.py:174]: I did it to restore it back to as it was before #9135. I agree it's probably better to leave it in this new way, which is more correct.

File: .github/SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Security
+
+See https://docs.wagtail.org/en/stable/contributing/security.html.
```
Comment (gasman) [inline .github/SECURITY.md:3]: I guess this should link to `/latest/`, given the note elsewhere.

**Comment by gasman** [comment]
Merged (with tweaks as commented above) in a2e1becc49620463620aeec56dfc15c5e85d10bd (main) and 9a2cff5a74597105161123c91e05ca8e2bdb0df9 (stable/5.0.x).

**Pull request closed without merging**